File: app/src/main/java/at/ac/tuwien/caa/docscan/ui/LogoutActivity.java
package at.ac.tuwien.caa.docscan.ui;

import android.content.Context;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import at.ac.tuwien.caa.docscan.R;
import at.ac.tuwien.caa.docscan.rest.LogoutRequest;
import at.ac.tuwien.caa.docscan.rest.User;

/**
 * Created by fabian on 24.08.2017.
 */

public class LogoutActivity extends BaseNavigationActivity implements LogoutRequest.LogoutCallback {


    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_logout);

        final Context context = this;

        Button confirmButton = (Button) findViewById(R.id.logout_confirm_button);
        confirmButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                // TODO: think about logout here:
                new LogoutRequest(context);
                User.getInstance().setLoggedIn(false);
                // Close the activity:
                finish();
            }
        });

        Button cancelButton = (Button) findViewById(R.id.logout_cancel_button);
        cancelButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                // Close the activity:
                finish();
            }
        });

    }

    @Override
    protected NavigationDrawer.NavigationItemEnum getSelfNavDrawerItem() {
        return NavigationDrawer.NavigationItemEnum.ACCOUNT_LOGOUT;
    }

    @Override
    public void onLogout() {
        User.getInstance().setLoggedIn(false);
    }
}

File: app/src/main/java/at/ac/tuwien/caa/docscan/ui/transkribus/logout/LogoutActivity.kt
package at.ac.tuwien.caa.docscan.ui.transkribus.logout

import android.content.Context
import android.content.Intent
import android.os.Bundle
import at.ac.tuwien.caa.docscan.databinding.ActivityLogoutBinding
import at.ac.tuwien.caa.docscan.ui.BaseNavigationActivity
import at.ac.tuwien.caa.docscan.ui.NavigationDrawer.NavigationItem
import org.koin.androidx.viewmodel.ext.android.viewModel

/**
 * Created by fabian on 24.08.2017.
 */
class LogoutActivity : BaseNavigationActivity() {

    companion object {
        fun newInstance(context: Context): Intent {
            return Intent(context, LogoutActivity::class.java)
        }
    }

    private val viewModel: LogoutViewModel by viewModel()
    private lateinit var binding: ActivityLogoutBinding

    override val selfNavDrawerItem = NavigationItem.LOGOUT

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityLogoutBinding.inflate(layoutInflater)
        setContentView(binding.root)
        binding.logoutConfirmButton.setOnClickListener {
            viewModel.logout()
            finish()
        }
        binding.logoutCancelButton.setOnClickListener {
            finish()
        }
    }
}
